.github/workflows/build-and-tag.yml

```diff
@@ -0,0 +1,35 @@
+name: Build and Tag
+on:
+  push:
+    branches:
+      - 'trunk'
```

Comment:
There have been _some_ cases where I've had to push to `trunk` twice if my local happened to somehow to be out of sync with `develop`, so pivoting this off pushes to `trunk` worries me a little but hopefully those are extreme edge cases (and perhaps I should do a better job ensuring I'm in sync).

Reply:
Does this [discussion](https://github.com/10up/ElasticPress/pull/2799/files#r881829479) relate (and also answer) to your concern? if not, we should brainstorm on the solution.